tests/std/tests/P0896R4_ranges_alg_copy_backward/test.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate that copy_backward_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::copy_backward_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<true>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<int*, int*>>);
+
+struct instantiator {
+    static constexpr int input[] = {13, 42, 1729};
+
+    template <class Bidi1, class Bidi2>
+    static constexpr void call() {
+        // For the second range, we need an iterator to the end; it's expedient to simply ignore ranges with differing
+        // iterator and sentinel types (i.e., ranges that don't satisfy common_range).
+        if constexpr (ranges::common_range<Bidi2>) {
+            using ranges::copy_backward, ranges::copy_backward_result, ranges::iterator_t;
+            const Bidi1 wrapped_input{input};
+
+            { // Validate range overload
+                int output[] = {-2, -2, -2};
+                Bidi2 wrapped_output{output};
+                auto result = copy_backward(wrapped_input, wrapped_output.end());

```

Comment:
We do not really test the one real reason we have for `copy_backward` namely overlapping regions. Should we test them? Especially for the memmove case

Reply:
Both `copy` and `copy_backward` handle overlapping regions, it's the direction of the allowed overlap that differs. Given:
```
|-----|-----|-----|
a     b     c     d
```
both `copy(b, d, a)` and `copy_backward(a, c, d)` are valid. 

The reason I don't have test coverage for overlapping ranges for `copy_backward` is the same reason we don't have test coverage for overlapping ranges for `copy`: both range algorithms are straightforward "rangifications" of our existing reliable `std` algorithm. In the case of `ranges::copy_backward` its literally implemented by the same lines of code. In cases where we reuse the `std` algorithm like this we've been performing fairly light coverage to ensure the adaptation is correct and reserving exhaustive testing of the underlying algorithm for cases where the range algorithm materially differs.

Reply:
Clarifying: this is my explanation for why there is no such test since you asked "Should we test them?" It's not a refusal to add such a test if you disagree with my rationale or simply think we should test it anyway.

Reply:
I'm ok with this light testing as long as the core algorithm is literally centralized. (e.g. for `reverse` I was nervous about not seeing empty/odd/variously-vectorized cases, because the code wasn't totally centralized)

I'd also be fine with more testing; almost seems easier to write than to talk about.

Reply:
Yeah I guessed it would be handled like that.  Sound absolutely reasonable to only check new implementations.

I guess when the TMP machinery is overhauled to allow `contiguous_iterator`s rather than plain pointers this would be the place to add tests

Reply:
Overlapping test added.